llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -249,7 +249,15 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
           .legalForTypesWithMemDesc({{s32, p0, s8, 8},
                                      {s32, p0, s16, 16},
                                      {s32, p0, s32, 32},
-                                     {p0, p0, sXLen, XLen}});
+                                     {p0, p0, sXLen, XLen},
```

Comment:
What about s16, s32, sXLen?

Reply:
Shouldn't be sXLen, but we do need s64 when ST.hasVInstructionsI64() is true.

nxv1s8 shouldn't be supported unless getELEN() == 64. Same as the type legality rules for G_ADD.